nixos/modules/system/boot/luksroot.nix

```diff
@@ -894,6 +896,19 @@ in
             '';
           };
         };
+
+        config = mkIf (clevis.enable && (hasAttr name clevis.devices)) {
+          preOpenCommands = mkIf (!systemd.enable) ''
+            mkdir -p /clevis-${name}
+            mount -t ramfs none /clevis-${name}
```

Comment:
@philiptaron Are you certain that `/run` is available at this step? Also, how do you handle the fact that the key could be leaked? `/run` continues to persist and it is not cleared before a long time.

Reply:
Setting up `/run` and co. is pretty much the [first thing that scripted initrd does](https://github.com/NixOS/nixpkgs/blob/faef749c0ea170adc9618838fcf58b144d2ae2fc/nixos/modules/system/boot/stage-1-init.sh#L149) (after setting up some shell functions), and systemd initrd does it before even starting generators.

But yes, we still very much should not use `/run` because A) it can leak to stage 2, and B) it can leak to swap in the worst of cases. `ramfs` cannot be swapped.

Reply:
@ElvishJerricco thanks for the detailed explanation, I did not have all these elements in mind.